### ISSUE TITLE
SSP support not available in this toolchain

### ISSUE THREAD

**Issue by Lukanite** [botfap/A95X-Linux-BuildEnv#1]
First of all, thanks for providing this code! This box will be a lot more useful once I can get Linux (hopefully Debian or Ubuntu one day) on it. However, when I try to run make using the A95X 2G defconfig (or any other defconfig), I run into the following error:

> > > toolchain-external undefined Configuring
> > > SSP support not available in this toolchain, please disable BR2_TOOLCHAIN_EXTERNAL_HAS_SSP
> > > package/pkg-generic.mk:185: recipe for target '/home/ubuntu/a95x/A95X-Linux-BuildEnv/buildroot/output/build/toolchain-external-undefined/.stamp_configured' failed
> > > make[1]: **\* [/home/ubuntu/a95x/A95X-Linux-BuildEnv/buildroot/output/build/toolchain-external-undefined/.stamp_configured] Error 1
> > > Makefile:36: recipe for target '_all' failed
> > > make: **\* [_all] Error 2

**Comment by corestaq** [comment]
Did you run "source TOOLSENV.sh" before running make?

What os? What release? X86 or X64? output of ls -la /opt/?

I recall having the same problem early on in developing it. I _think_ it was because I was missing 32bit dev libs on a 64 bit platform.

My additional packages from stock ubuntu 14.04 server:
sudo dpkg --add-architecture i386
sudo apt-get update
sudo apt-get upgrade
sudo apt-get install p7zip-full p7zip-rar squashfs-tools micro-httpd subversion zip unzip git tree libncurses5-dev bzip2 lib32z1 lib32gcc1 build-essential
sudo apt-get install libc6:i386 libncurses5:i386 libstdc++6:i386

**Comment by corestaq** [comment]
Ive pushed a new V0.1 release with new toolchain locations and build env script, please try

**Comment by corestaq** [comment]
OK so just come across this issue with another user.

Was a problem with the external toolchains conflicting with system installed toolchains and git missing a dot file that was needed for config

Ive pushed out some updates to fix it and its fixed in release V0.2 and in Master

**Comment by Lukanite** [comment]
Thanks for the update! The build process works now on 16.04 LTS x64. However, I've tried setting up an SD card for the NEXBOX A95X S905X 2G/16G, but I can't get the NEXBOX to boot off of it - trying to boot into recovery just lands me at the Android recovery. Any ideas?

**Comment by corestaq** [comment]
Yes. I left that part out of the instructions on purpose because it can cause bricking.

When the S905X CPU initialises it looks for the bootloader first on the emmc then the sd then the usb. To get it to use the bootloader on the sd you have to erase the bootloader on the emmc to get the cpu to branch to its next boot option.

Take a backup of your emmc boot load and fat:
from a root terminal: dd if=/dev/mmcblk0 of=stock-uboot.bin bs=1M count=1
save that file somewhere safe on a usb or sd.

Wipe out the existing bootloader:
dd if=/dev/zero of=/dev/mmcblk0 bs=1k count=512

Without sd card you should now get a red light at power on and fail to boot. If you insert your uboot prepped sd card now you can boot from it.

To restore the stock boot loader afterwards:
dd if=stock-uboot.bin of=/dev/mmcblk0 bs=1M count=1

**Comment by Lukanite** [comment]
Alright, thanks again for the awesome info; I'll try that out shortly. By the way, the folks over at [https://forum.libreelec.tv/thread-1176-page-26.html](LibreElec) seem to have figured out a way to use aml_autoscript on the SD card to cause an SD card boot without having to overwrite the internal bootloader. If you have the chance, perhaps using that will make things a bit less risky?

**Comment by Lukanite** [comment]
Whew, looks like I didn't brick my device! I've confirmed that the SD bootloader is working since I have the red light with no SD inserted and a blue light with it in. However, when I have the SD card in, the system just boots to Android.

I think it's a problem with the command `env default -a env save run bootsdcard`, which just yielded something along the lines of `default: no such file or directory`. Touching a "default" file didn't seem to do anything, either.

Any ideas?

**Comment by Lukanite** [comment]
Never mind, I'm just an idiot. After accessing the serial console (the top pin is ground, then RX below that, then TX below that) and holding enter to enter uboot, I was able to run the env commands successfully, and now I have it booting from the SD card by default. 

Thanks again for all of your help and work!

**Comment by corestaq** [comment]
Re-opening as reminder for V0.3 stock bootloader updates

**Comment by LeosSire** [comment]
Hello, I know its been a few years, but do you have a copy of TOOLSENV.sh? 

This has been removed from amlogic's server.

FYI - All the other packages have been moved to:
http://openlinux.amlogic.com:8000/download/GPL_code_release/ThirdParty/
